src/Configuration/Levels/LevelRulesResolver.php

```diff
@@ -16,11 +16,14 @@ final class LevelRulesResolver
      */
     public static function resolve(int $level, array $availableRules, string $methodName): array
     {
+        Assert::natural($level);
```

Comment:
line 31-32 then can be removed, or just moved here:

```suggestion
        // level < 0 is not allowed
        Assert::natural($level, sprintf('Level must be >= 0 on %s', $methodName));
```